src/v/cluster/metadata_cache.cc

```diff
@@ -179,6 +179,8 @@ std::optional<model::node_id> metadata_cache::get_controller_leader_id() {
     return _leaders.local().get_leader(model::controller_ntp);
 }
 
+void metadata_cache::reset_leaders() { _leaders.local().reset(); }
```

Comment:
Please describe in the commit message (1) how/why/when this reset should be invoked and (2) what the expected outcome is, such as what would would happen to metadata request that occurred right after this and how does the cache become repopulated and what is that latency.